commit 031516123d25a12808ab4e56ea8deda74946cdcd
Author: Koen Kooi <koen@dominion.thruhere.net>
Date:   Wed Jan 26 12:04:36 2011 +0100

    Revert "Set a large CRTC upper limit to not prune larger resolutions"

    Picture is garbled after switching resolutions, so revert it.

    Virtual size too big, revert the commit 9c4d7592dcb7dc20a48a6f941d9d94bd73d34153.

	Upstream-Status: Pending

diff --git a/src/omapfb-crtc.c b/src/omapfb-crtc.c
index 9aaa52f..cbeff35 100644
--- a/src/omapfb-crtc.c
+++ b/src/omapfb-crtc.c
@@ -190,13 +190,13 @@ OMAPFBCRTCInit(ScrnInfoPtr pScrn)
 	 * In practise, this doesn't seem to be supported.
 	 * (no way to setup the overlay offset/base address)
 	 */
-	 /* FIXME: figure out what makes sense here. A known max resolution?
-	  * framebuffer size?
-	  */
 	xf86CrtcSetSizeRange(pScrn,
-	                     8, 8, 2048, 2048);
+	                     8, 8,
+	                     ofb->state_info.xres_virtual,
+	                     ofb->state_info.yres_virtual);
 
 	ofb->crtc = xf86CrtcCreate(pScrn, &OMAPFBCrtcFuncs);
+
 }
 
 
